Source/Laser/LaserParticleContainer.H

```diff
@@ -1,13 +1,13 @@
 #ifndef WARPX_LaserParticleContainer_H_
 #define WARPX_LaserParticleContainer_H_
 
-#include <limits>
+#include <LaserProfiles.H>
 
 #include <WarpXParticleContainer.H>
 #include <WarpXConst.H>
 #include <WarpXParser.H>
 
-enum class laser_t { Null, Gaussian, Harris, parse_field_function };
+#include <memory>
```

Comment:
Limits is still needed in this file:
```suggestion
#include <memory>
#include <limits>
```